Porting core p7 2

jkeenan opened the pull request:
@atoomic, since these commits entailed running various `regen` scripts, they will need more careful review.

jkeenan committed:
Regenerate database used in t/porting/customized.t

jkeenan committed:
Add placeholder KVP for 7.000000 in %delta

%delta gets transformed into %version, which is checked by
t/porting/corelist.t.

$Module::CoreList::VERSION will have to be updated later (presumably
when we update all $VERSIONs).

jkeenan committed:
De-prototype one sub

... at least until we figure out how prototypes will work.

Restore one '5' from a '7'; we really meant '5.14' in this case.

jkeenan committed:
Store results of running various regen/*.pl programs

Including, but not limited to:

./perl -Ilib regen/regen_lib.pl
./perl -Ilib regen/mk_invlists.pl
./perl -Ilib regen/regcharclass.pl
./perl -Ilib regen/embed.pl

This gets us closer to having t/porting/regen.t PASS.  What's still
failing:

ok - regen/pX.pl lib/p5.pm
'lib/p7.pm-new' and 'lib/p7.pm' differ
not ok - regen/pX.pl lib/p7.pm
'perl7.h-new' and 'perl7.h' differ
not ok - regen/pX.pl perl7.h
ok - regen/unicode_constants.pl unicode_constants.h

Parse errors: Bad plan.  You planned 46 tests but ran 48.

jkeenan committed:
Avoid uninitialized value warning, which caused subsequent failure

atoomic commented on t/porting/copyright.t:
```diff
@@ -27,6 +27,7 @@ if ( $Config{usecrosscompile} ) {
 }
 
 my ($opt) = @ARGV;
+$opt ||= '';
```
this could probably use safely `//=` instead as this is a porting tool?

jkeenan replied:
Okay, see 2021c39cbb5f598a82bcd552898cb09c3a0690f3

atoomic commented on regen/embed.pl:
```diff
@@ -46,7 +46,7 @@ ($)
 }
 
-sub full_name ($$) { # Returns the function name with potentially the
+sub full_name { # Returns the function name with potentially the
```
alternatively we could consider using `sub full_name :prototype($$)`

jkeenan replied:
Since I've never used prototypes at all, I'll have to trust you on that!  See 7aad926102

jkeenan committed:
Get t/porting/test_bootstrap.t to pass

Note:  It was failing because it detected two instances of 'use ' in
t/comp/redef.t.  While the insertion of 'use p5; use warnings;' might be
necessary here, it goes against the principle (which test_bootstrap.t
was exercising) of not using 'use ' statements in t/comp/*.t -- the
oldest and most basic tests of perl's functionality.

jkeenan committed:
Per atoomic, defined-or better here

jkeenan committed:
Can use ':prototpe' syntax here, per atoomic

jkeenan committed:
Make test run correctly under strictures

t/porting/ss_dup.t was never converted to 'use strict;' in Perl 5.
Declare variables with 'my' to pass under strict.

jkeenan committed:
lib/diagnostics.t was never converted to 'use strict;'

Do so now.

jkeenan commented:
@atoomic This is where I'll leave off for today.  Please review and if okay, merge into core-p7.  Thx.
